File: app/src/main/java/com/expensivebelly/dagger2retaingraph/activity/di/RetainGraphComponent.java
package com.expensivebelly.dagger2retaingraph.activity.di;

import com.expensivebelly.dagger2retaingraph.activity.RetainGraphActivity;

import dagger.Component;

@ActivityScope
@Component(modules = RetainModule.class)
public interface RetainGraphComponent {

    void inject(RetainGraphActivity retainGraphActivity);

    final class Initializer {
        private Initializer() {
        } // No instances.

        public static RetainGraphComponent init() {
            return DaggerRetainGraphComponent.builder().build();
        }
    }
}

File: app/src/main/java/com/expensivebelly/dagger2retaingraph/activity/di/RetainGraphComponent.kt
package com.expensivebelly.dagger2retaingraph.activity.di

import com.expensivebelly.dagger2retaingraph.activity.MainActivity
import dagger.Component

@ActivityScope
@Component(modules = [RetainModule::class])
interface RetainGraphComponent {
    fun inject(mainActivity: MainActivity)
}